pkgs/development/tools/mcap/mcap-cli/default.nix

```diff
@@ -0,0 +1,32 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+}:
+
+buildGoModule rec {
+  pname = "mcap-cli";
+  version = "0.0.28";
+
+  src = fetchFromGitHub {
+    owner = "foxglove";
+    repo = "mcap";
+    rev = "fd3ecada218f41929c6d430acba3561e5ebfaf08";
+    sha256 = "sha256-pIZThgb0wRmhNi3YrYTC+Tr6a/4Q7CWuHw+icjb1p8g=";
+  };
+
+  sourceRoot = "source/go/cli/mcap";
+
+  vendorHash = "sha256-3TSwOgTIaBg5U8UdQ74LmsS8gS05Yw02JjfaIN9fKQQ=";
+
+  GOWORK = "off";
+
+  # tests are currently failing because the test mcap files are in another directory in the repo
```

Comment:
These tests are still failing because they refer to paths in relative directories and even using `sourceRoot` doesn't seem to fix this
https://github.com/foxglove/mcap/blob/main/go/cli/mcap/cmd/cat_test.go#L23

Reply:
It looks like they're expecting to be run from go/cli/mcap/cmd?  What if we changed into that directory in `preCheck`, and changed back in `postCheck`?

Reply:
It looks like this test is actually just broken in upstream mcap:

https://github.com/foxglove/mcap/issues/895